### ISSUE TITLE
Displaying anything from an ATL block using the 'with' statement makes it forget it's yoffset

### ISSUE THREAD

**Issue by Galo223344** [renpy/renpy#4867]
As the tile says, whenever an ATL block displays a LayeredImageProxy, it'll forget it's yoffset.

I'm using the LayeredImageProxy for flipping the sprites, like so

```
image dude left = LayeredImageProxy("dude", Transform(xzoom=-1.0))
image dude right = LayeredImageProxy("dude", Transform(xzoom=1.0))
```

Whenever I change the orientation of the sprite from an ATL block like this
```
show dude:
    "dude left" with Dissolve(1)
```
The sprite will lose it's yoffset causing it to shift down. But if I do this instead
```
show dude:
    "dude left"
```
The sprite will retain it's yoffset.

This does not happen if I apply an attribute to the layeredimage or if I flip the image outside of an ATL block (Even if I use 'with Dissolve')

Below I have attached a minimal reproducible example.
[Testing.zip](https://github.com/renpy/renpy/files/12142206/Testing.zip)

**Comment by Gouvernathor** [comment]
There is the complex behavior of [transform replacement](https://www.renpy.org/dev-doc/html/atl.html#replacing-transforms) to take into account, to figure out which is the expected behavior.
Worth noting, the offset is applied as ATL properties in the layeredimage body. I suspect the bug has little to do with layeredimage or layeredimageproxy, an that any transformed displayable may be subject to similar behavior. I think it's more to do with transitions in ATL.

**Comment by Gouvernathor** [comment]
As suspected, doing this :
```rpy
show a:
    "a" with Dissolve(1)
```
using both an auto-defined normal image and the following wrapper : `image postest = Transform(yoffset=10)` results in the same thing : the offset is ignored.
So, as suspected, this is not a layeredimage problem.

> This does not happen if I apply an attribute to the layeredimage or if I flip the image outside of an ATL block (Even if I use 'with Dissolve')

By "this does not happen", you mean that the problematic behavior, which is offset being ignored, does not trigger in any situation other than with-inside-ATL ?

**Comment by Galo223344** [comment]
Exactly. It also doesn't happen if I don't include the 'with' statement
inside the ATL block.

So if I do
show a:
    "a"

The yoffset won't be ignored

**Comment by Gouvernathor** [comment]
Found the issue, this is a Dissolve issue (other transitions may also have it, but a custom ATL transition I tested with doesn't have it, so...).
Inside the Dissolve render, the offset is accessible : `renpy.exports.get_placement(self.new_widget).yanchor` is 10, in my "postest" example. But the render algorithm disregards that placement, and renders both children at 0, 0.
I suppose it should be calculated and applied _after_ the old and new widgets are rendered, but I'm not certain how. @renpytom ? [Related code](https://github.com/renpy/renpy/blob/54b44069740d628161d6d652774b3b46264e8cbf/renpy/display/transition.py#L403)
